docs/advanced-features/event-driven-packaging.md

```diff
@@ -0,0 +1,298 @@
+for additional manual or automatic processing.
+
+EDP, upon completion and if configured to do so, may warm its
+contents to a File Gateway where it has read permissions to ensure
+that new EDP-created Quilt packages are available to Gateway clients
+like Windows Workspaces.
+
+> Users can optionally subscribe directly to the EDP SNS topic.
+This is useful for both debugging and viewing how events are structured.
+
+## Example: Lambda function to automatically create data packages
+
+1. An instrument automatically uploads a folder containing files
+from a single experiment into
+_s3://instrument-bucket/instrument-name/experiment-id/_.
+1. EDP listens for events in
+_s3://instrument-bucket/instrument-name/experiment-id/*_.  After the
+specified duration or event count, a `package-objects-ready` event
+is generated and sent to EventBridge.
+1. A custom SNS topic is created for monitoring data package creation
+that Lab and Computational scientists subscribe to (`SNS_TOPIC_ARN`).
+1. A custom lambda fuction triggered by the `package-objects-ready`
+event processes the experiment files and generates a data package.
+Additional processing includes (but is not limited to):
+    - Decorating the data package with helpful additional files:
+        - `README.md`: Noting that the package was created by EDP,
+	a custom lambda function, and validated with a [Quilt
+	workflow](../advanced-features/workflows.md).
+        - [`quilt_summarize.json`](../Catalog/VisualizationDashboards.md#quilt_summarize.json)
+        - [`.quiltignore`](../advanced-features/.quiltignore.md)
+    - Package metadata creation and validation: Send an SNS
+    notification on [metadata
+    validation](../advanced-features/workflows.md) failure.
+
+    <!--pytest.mark.skip-->
+    ```python
+    import datetime
+    import functools
+    import os
+    import pathlib
+    import tempfile
+    import boto3
+    import quilt3 as quilt3
+    from aws_lambda_powertools import Logger
+
+    logger = Logger()
+    s3 = boto3.client("s3")
+    sns = boto3.client("sns")
+
+    # Configuration environment variables defined for Lambda function
+    WORKFLOW_NAME = os.environ.get("WORKFLOW_NAME") or ...
+    QUARANTINE_BUCKET_NAME = os.environ["QUARANTINE_BUCKET_NAME"]
+    SNS_TOPIC_ARN = os.environ["SNS_TOPIC_ARN"]
+    MAX_SNS_SUBJECT_LEN = 99
+    QUILT_URL = os.environ["QUILT_URL"]
+
+    # Empty quilt_summarize.json object
+    QUILT_SUMMARIZE_JSON_STR = "[]"
+
+    # README.md default Markdown
+    QUILT_README_STR = f"""#Quilt package auto-generated by EDP\n\n
+    Created on {datetime.date.today()} by an 
+    automated Lambda agent for the {WORKFLOW_NAME} workflow."""
+
+    # File system files for Quilt to ignore
+    QUILT_IGNORE_STR = """.DS_*
+    Icon
+    ._*
+    .TemporaryItems
+    .Trashes
+    .VolumeIcon.icns
+    """
+
+    # Define helpful additional data package files
+    beautify_files = {
+        "quilt_summarize.json": QUILT_SUMMARIZE_JSON_STR,
```

Comment:
What is the point of an empty quilt_summarize? Eliminate unless you make use of it. README.md is not a good candidate as that automatically displays.

Reply:
A `quilt_summarize.json` file is considered a "standard file" for all data packages. We cannot dynamically create content for the quilt_summarize unless we know what is going to be in the S3 bucket. I don't think that means that an example shouldn't include it because we still get questions from customers asking what it is and how it works. I can create a dummy file for the purposes of this example script